lib/main.js

```diff
@@ -55,14 +67,56 @@ export default {
           });
         }
       }),
-    );
-
-    this.subscriptions.add(
       atom.config.observe('linter-tslint.useLocalTslint', (use) => {
         tslintCache.clear();
         this.useLocalTslint = use;
       }),
     );
+
+    atom.workspace.observeActivePaneItem((activeItem) => {
+      if (activeItem && activeItem.getGrammar) {
+        if (isTypescriptGrammar(activeItem.getGrammar())) {
```

Comment:
Instead of using a non-public API can we validate it based on scopes like everything else with linter providers is done?

You can find an example of how to use the cursors to get the scopes and check it using the public API from `linter-eslint` [here](https://github.com/AtomLinter/linter-eslint/blob/bd2e81f176f81027a2b94a55a3310bf9b0a4ab97/src/main.js#L51-L53).